scipy/stats/tests/test_distributions.py

```diff
@@ -3643,6 +3643,44 @@ def test_ppf(self, p, a, b, expected):
         x = stats.betaprime.ppf(p, a, b)
         assert_allclose(x, expected, rtol=1e-14)
 
+    @pytest.mark.parametrize(
+        'x, a, b, expected',
+        [(1e22, 0.05, 0.1, 0.9978811466052919),
+         (1e22, 100.0, 0.05, 0.8973027435427167),
+         (1e10, 0.05, 0.1, 0.9664184367890859),
+         (1e10, 1.5, 1.5, 0.9999999999999983),
+         (1e10, 100.0, 0.05, 0.5911548582766262),
+         (1e-10, 0.05, 0.1, 0.21238845427095),
+         (1e-10, 1.5, 1.5, 1.6976527260079727e-15),
+         (1e-10, 0.05, 100.0, 0.4088451417233739),
+         (1e-22, 0.05, 0.1, 0.05334956764928733),
+         (1e-22, 1.5, 1.5, 1.6976527263135503e-33),
+         (1e-22, 0.05, 100.0, 0.10269725645728332),
+         (1e-100, 0.05, 0.1, 6.716312642191984e-06),
+         (1e-100, 1.5, 1.5, 1.6976527263135503e-150),
+         (1e-100, 0.05, 100.0, 1.292881858756166e-05)])
+    def test_cdf_tails(self, x, a, b, expected):
+        # test for gh-17631
+        # values computed with mpmath. Example:
+        # import mpmath as mp
+        # mp.dps = 25
+        # a, b = mp.mpf('0.05'), mp.mpf('0.1')
```

Comment:
Warren pointed this out before: for decimal numbers that cannot be represented exactly in the NumPy floating point format, it is not technically appropriate to use `mpf` on strings because it gives `mpmath` and the SciPy function different *inputs*. For this reason, even if `0.05` can be represented exactly by NumPy, I think it would be preferable to do just `mp.mpf(0.05)` so as to avoid giving the impression that using quotes is the "right" thing to do.

Reply:
I'll take a look at the rest later.

Reply:
I haven't thought about it this way. I aimed to get the "most accurate" value possible with dps=50 and then round to float at the end. in the end, it does not seem to make much of a difference, I can commit your suggestions below later